webClient/src/app/editor/code-editor/code-editor.component.ts

```diff
@@ -97,7 +100,17 @@ export class CodeEditorComponent implements OnInit, OnDestroy {
         }
       }
       this.openFile(fileNode);
-      this.editorControl.editor.getValue().layout();
+      this.editorControlService.updateBottomBar.next(fileNode);
+      this.editorControl.editor.getValue().layout();      
+    });
+
+    this.editorControlService.updateBottomBar.subscribe((fileNode: ProjectStructure) => {
+     
+      
+      
+      document.getElementById('encoding').innerHTML = editorControlService.getStringEncoding(fileNode.encoding);
+      document.getElementById('ext').innerHTML = editorControlService.GetExtInfo();
+      document.getElementById('lang').innerHTML = editorControlService.GetLangInfo();
```

Comment:
This is okay but it's 1. outdated 2. only works on browsers; Angular has a cleaner way to do this. Check out:

ViewChild & ElementRef in zlux-editor\webClient\src\app\app.component.ts
It finds & manipulates elements based on #id attribute in the HTML

Looks like we still do it this JS way in very old code from zlux-editor\webClient\src\app\shared\editor-control\editor-control.service.ts but there's not a single usage of the method itself